scala-package/core/src/main/scala/org/apache/mxnet/FeedForward.scala

```diff
@@ -224,13 +224,24 @@ class FeedForward private(
     var i = 0
     while (data.hasNext && i != numBatch) {
       val batch = data.next()
-      i += 1
-      ExecutorManager.loadData(batch, dataArrays)
-      predExec.forward(isTrain = false)
-      val padded = batch.pad
-      val realSize = batchSize - padded
-      for ((list, nd) <- outputs zip predExec.outputs) {
-        list += nd.slice(0, realSize).copy()
+      try {
```

Comment:
I find the try finally really annoying and redundant. MXNet actually seg-fualts for simple cases such as the input shape is wrong. We are adding these try-finally and making the code more complicated